arkouda/dataframe.py

```diff
@@ -1841,6 +1844,269 @@ def numeric_help(d):
+                columns[colName] = Categorical.attach(name)
+            elif f"_{SegArray.objtype}_" in name:
+                columns[colName] = SegArray.attach(name)
+
+        index_resp = cast(str, generic_msg(cmd="attach", args=f"df_index_{user_defined_name}_key"))
+        dtype = index_resp.split()[2]
+        if dtype == Strings.objtype:
+            ind = Strings.from_return_msg(index_resp)
+        else:  # pdarray
+            ind = create_pdarray(index_resp)
+
+        index = Index.factory(ind)
+
+        df = DataFrame(columns, index)
+        df.name = user_defined_name
+        return df
+
+    @staticmethod
+    @typechecked
+    def unregister_dataframe_by_name(user_defined_name: str) -> None:
+        """
+        Function to unregister DataFrame object by name which was registered
+        with the arkouda server via register()
+
+        Parameters
+        ----------
+        user_defined_name : str
+            Name under which the DataFrame object was registered
+
+        Raises
+        -------
+        TypeError
+            if user_defined_name is not a string
+        RegistrationError
+            if there is an issue attempting to unregister any underlying components
+
+        See Also
+        --------
+        register, unregister, attach, is_registered
+        """
+
+        registry = list_registry()
+        matches = []
+        regEx = compile(
+            f"^df_data_[a-zA-Z0-9]+_({pdarray.objtype}|{Strings.objtype}|"
+            f"{Categorical.objtype}|{SegArray.objtype})_{user_defined_name}"
+        )
+        # Using the regex, cycle through the registered items and find all the columns in the DataFrame
+        for name in registry:
+            x = match(regEx, name)
+            if x is not None:
+                matches.append(x.group())
+        matches.sort()
+
+        if len(matches) == 0:
+            raise RegistrationError(f"No registered elements with name '{user_defined_name}'")
+
+        # Remove duplicates caused by multiple components in categorical or segarray
+        dedupe = list(dict.fromkeys(matches))
+
+        for name in dedupe:
+            if f"_{Strings.objtype}_" in name or f"_{pdarray.objtype}_" in name:
+                cols_resp = cast(str, generic_msg(cmd="attach", args=name))
+                dtype = cols_resp.split()[2]
+                if dtype == Strings.objtype:
+                    Strings.unregister_strings_by_name(name)
+                else:  # pdarray
+                    unregister_pdarray_by_name(name)
+            elif f"_{Categorical.objtype}_" in name:
+                Categorical.unregister_categorical_by_name(name)
+            elif f"_{SegArray.objtype}_" in name:
+                # SegArray does not have an unregister_by_name function. By attaching and using
+                # unregister() it prevents unregistering the components individually here
+                segarray = SegArray.attach(name)
+                segarray.unregister()
```

Comment:
My brain does not like the idea of attaching for the purpose of unregistering, which seems to happen in a few of the cases. I believe the first block could be reorganized to something like
```python
if f"_{Strings.objtype}_" in name:
    Strings.unregister_strings_by_name(name)
elif f"_{pdarray.objtype}_" in name:
    unregister_pdarray_by_name(name)
```
Since we are in the process of moving SegArray to the server, this is totally fine for now because it will most likely require updates for that case in the near future.

Reply:
This was reused logic from the attach case that isn't needed here - good catch

Reply:
Do we have an issue in to add an unregister by name to segarray?